app/handlers/ErrorHandler.scala

```diff
@@ -16,20 +16,34 @@
 
 package handlers
 
+import config.FrontendAppConfig
 import play.api.i18n.{I18nSupport, MessagesApi}
-import play.api.mvc.Request
+import play.api.mvc.Results.Ok
+import play.api.mvc.{Request, RequestHeader, Result}
 import play.twirl.api.Html
+import uk.gov.hmrc.http.ExpectationFailedException
 import uk.gov.hmrc.play.bootstrap.frontend.http.FrontendErrorHandler
-import views.html.ErrorTemplate
+import views.html.{ErrorTemplate, NoDataEnteredErrorView}
 
 import javax.inject.{Inject, Singleton}
+import scala.concurrent.Future
 
 @Singleton
 class ErrorHandler @Inject()(
```

Comment:
I'm not sure it's a good idea to put this redirection into here. This is a generic error handler and if any other part of the service threw same exception they would be redirected to this very specific error page. I would redirect to this error page on the declaration controller instead.

Reply:
@PavelVjalicin Do you have any thoughts on this?

Reply:
My thoughts is that we will introduce more precise error handling in the future as we see fit, alternatively we can do that now.

Reply:
I think we'll just introduce a custom exception for this use case and handle it.

Reply:
Normally for this sort of thing we just redirect in the controller rather than throwing then catching an exception. Not sure which is the best way but we may end up with a lot of exception classes if we go this route

Reply:
@grantrobson I think in this case throwing an exception is fine. Since this is not part of a regular Journey, this happens only via exceptional conditions. I don't want to add a bunch of code that is not related to the journey for an edge case scenario instead of bypassing everything with a simple exception throw. 

This exception might also be useful for other instances where this is appropriate.

Reply:
OK, I'm fine with this then. I will approve PR once Bonnie creates the exception